src/main/scala/mesosphere/marathon/core/groupmanager/impl/GroupManagerActor.scala

```diff
@@ -0,0 +1,279 @@
+package mesosphere.marathon.core.groupmanager.impl
+
+import java.net.URL
+import javax.inject.Provider
+
+import akka.actor.{ Actor, ActorLogging, Props }
+import akka.event.EventStream
+import akka.pattern.pipe
+import mesosphere.marathon._
+import mesosphere.marathon.api.v2.Validation._
+import mesosphere.marathon.core.event.{ GroupChangeFailed, GroupChangeSuccess }
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.io.PathFun
+import mesosphere.marathon.io.storage.StorageProvider
+import mesosphere.marathon.state.{ AppDefinition, Container, PortDefinition, _ }
+import mesosphere.marathon.upgrade.{ DeploymentPlan, GroupVersioningUtil, ResolveArtifacts }
+import mesosphere.util.CapConcurrentExecutions
+import org.slf4j.LoggerFactory
+
+import scala.collection.immutable.Seq
+import scala.collection.mutable
+import scala.concurrent.Future
+import scala.util.{ Failure, Success }
+
+private[groupmanager] object GroupManagerActor {
```

Comment:
While the pipeTo is nice, it seems a bit off to me- as it stands there is little reason to actually have the GroupManager as an actor since everything it does is purely future based, so the Single Threaded Apartment has almost no value.

Instead, for every `Request`, there should be a corresponding `Response` message type (unless its "Done") and the Future methods should change the actor's internal state while they are being processed. I think the latter is out of the scope of this change, but it would be good to adapt a `Response` type and use it.

This is mostly a cosmetic comment, but there is some value in it. Perhaps, simply adding "Replies with T" to all of the request types would help?